### PR TITLE
Add an section for developers in the patch note

### PR DESCRIPTION
**Pull request by Bouh**
We can add a spoiler at the bottom of the patch note for keeping a track of important thing for devs.

```
<details>
  <summary>For developpers</summary>
  
  Spoiler text extract from commits like this one:
https://github.com/4ian/GDevelop/commit/cc158a9250f8c275a590253e5f1f9e28d8ddf3c4
And others
 
</details>
```

<details>
  <summary>For developpers</summary>
  
  Spoiler text extract from commits like this one:
https://github.com/4ian/GDevelop/commit/cc158a9250f8c275a590253e5f1f9e28d8ddf3c4
And others
 
</details>

Merge pull request  has no style.
image

### CODE REVIEW

File: newIDE/app/scripts/extract-changelog.js [hunk 1]
```diff
@@ -183,8 +185,9 @@ const formatHiddenCommitMessage = commit => {
 
   const hiddenCommits = commitsWithAuthors.filter(commit => commit.hidden);
   const displayedCommits = commitsWithAuthors.filter(commit => !commit.hidden);
+  const devCommits = displayedCommits.filter(commit => commit.forDev);
   const fixCommits = displayedCommits.filter(commit => commit.isFix);
-  const improvementsCommits = displayedCommits.filter(commit => !commit.isFix);
+  const improvementsCommits = displayedCommits.filter(commit => { return !commit.isFix && !commit.forDev});
```
Comment (4ian) [inline newIDE/app/scripts/extract-changelog.js:190]: nitpicking:

```suggestion
  const improvementsCommits = displayedCommits.filter(commit => !commit.isFix && !commit.forDev);
```

File: newIDE/app/scripts/extract-changelog.js [hunk 2]
```diff
@@ -199,4 +202,11 @@ const formatHiddenCommitMessage = commit => {
 
   shell.echo(`\n## 🐛 Bug fixes\n`);
   shell.echo(fixCommits.map(formatCommitMessage).join('\n'));
+
+  if (devCommits) {
+    shell.echo(`\n<details>\n`);
+    shell.echo(`\n<summary> 🛠 Developers</summary>\n`);
```
Comment (4ian) [inline newIDE/app/scripts/extract-changelog.js:208]: We'll have to test how this renders inside GDevelop when showing the changelog

(if you want to try, launch Storybook and search the `Changelog` component. Modify the code in `GDevelopServicesTestData.js` to insert this and see how it renders)

**Comment by 4ian** [comment]
> Merge pull request has no style

For the `Merge pull request #`, we should hide these commits (because all the information is inside the other commits of the branch that was merged).

**Comment by Bouh** [comment]
The spoiler is useable :) 

image

File: newIDE/app/scripts/extract-changelog.js [hunk 3]
```diff
@@ -201,4 +204,13 @@ const formatHiddenCommitMessage = commit => {
 
   shell.echo(`\n## 🐛 Bug fixes\n`);
   shell.echo(fixCommits.map(formatCommitMessage).join('\n'));
+
+  if (devCommits) {
+    shell.echo(`\n<details>`);
+    shell.echo(`<summary> 🛠 Developers</summary>`);
```
Comment (arthuro555) [inline newIDE/app/scripts/extract-changelog.js:210]: You should not use HTML as if I remember correctly the changelog is extracted to be put on GitHub. GitHub will escape the HTML for obvious security reasons, and there will just be the ugly tags polluting the changelog without their effect. Just use markdown ;)

Comment (Bouh) [reply newIDE/app/scripts/extract-changelog.js:210]: This HTML work in GDevelop and github :)

Comment (arthuro555) [reply newIDE/app/scripts/extract-changelog.js:210]: It seems so, nevermind about the fact that GitHub escapes it, but it still is a security issue. Imagine someone makes a MitM attack and adds malicious scripts to the changelog, it wouldn't be escaped by GDevelop and give remote code execution on an environment with node, that is very dangerous!
Maybe GDevelop could replace those specific HTML tags with a Markdown version and use a plugin for the markdown parser to add spoilers.

Comment (arthuro555) [reply newIDE/app/scripts/extract-changelog.js:210]: Here is a remark (the used markdown renderer) plugin that would do that: https://github.com/Rokt33r/remark-collapse

Comment (4ian) [reply newIDE/app/scripts/extract-changelog.js:210]: We should indeed not accept arbitrary html. Let me know if you can use this or something else to accept **only** summary and details :)

**Comment by Bouh** [comment]
Nothing in the proposals is working.
If I escape the HTML then it's useless to detect or not detect details/summary tags.

There is a way to detect some markdown node types like it's done [here](makeMarkdownCustomRenderers) . 
But since the spoiler is not supported by markdown or by a pure HTML component it doesn't seem possible.

**Comment by arthuro555** [comment]
What I was saying is working I think?
For the github leave it like currently. For the IDE use the remeark plugin and a regex or something to replace the html with mardown equivalent.

**Comment by Bouh** [comment]
There is no equivalent in md.

**Comment by arthuro555** [comment]
For the 3rd time, there is if you use a remark plugin. I sent you the link to that plugin before: https://github.com/Rokt33r/remark-collapse

**Comment by Bouh** [comment]
Yes, I'm talking about the IDE only.
So what, i use remark plugin for parse the extracted changelog, find` <details>` and` <summary>` and the result string in output contain also` <details>` and `<summary>` and i keep `escapeHTML={false}` ?
I must be missing the logic somewhere for me it's the same.

**Comment by Bouh** [comment]
The complexity of filtering tags is not worth it compared to just putting the info in a smaller category.
It's like Arthuro suggested in PM, add the Dev category in a H3.

**Comment by 4ian** [comment]
I agree! Let's not complicate our life too much and stay on the "safe and simple" side of things. Looks good to me! :)

**Pull request merged**